plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java

```diff
@@ -188,9 +208,27 @@ public void put(final MavenPath path, final Payload payload)
     try (StorageTx tx = getStorage().openTx()) {
       if (path.getCoordinates() != null) {
         putArtifact(path, payload, tx);
+        if (payload instanceof MavenPayload) {
```

Comment:
This part goes away if we get https://github.com/sonatype/nexus-oss/pull/1144 in place, I dislike it too very much